src/librbd/cache/ParentCacheObjectDispatch.cc

```diff
@@ -45,8 +45,11 @@ void ParentCacheObjectDispatch<I>::init(Context* on_finish) {
   auto cct = m_image_ctx->cct;
   ldout(cct, 5) << dendl;
 
-  if (m_image_ctx->parent != nullptr) {
+  if (m_image_ctx->child == nullptr) {
     ldout(cct, 5) << "child image: skipping" << dendl;
```

Comment:
nit: may be change the log message to "not parent image: skipping"? Because a child may be another parent...